torch/onnx/_backend/core.py

```diff
@@ -0,0 +1,852 @@
+import dataclasses
+import importlib
+import logging
+
+from functorch.compile import min_cut_rematerialization_partition
+
+import torch
+import torch._C
+import torch._ops
+import torch._prims.executor
+import torch.fx
+import torch.onnx
+import torch.onnx._internal
+import torch.onnx._internal.diagnostics
+import torch.onnx._internal.exporter
+import torch.onnx._internal.fx.decomposition_table
+import torch.onnx._internal.fx.passes
+from torch._dynamo.backends.common import aot_autograd
+from torch._subclasses.fake_tensor import FakeTensor
+from torch.fx.passes.fake_tensor_prop import FakeTensorProp
+from torch.fx.passes.infra.partitioner import CapabilityBasedPartitioner
+from torch.fx.passes.operator_support import OperatorSupport
+from torch.fx.passes.tools_common import CALLABLE_NODE_OPS
+from torch.onnx._internal.exporter import ExportOptions
+from torch.utils import _pytree
+
+try:
+    # Use try-except to initialize package-dependent global variables.
+    from typing import Any, Dict, List, Mapping, Optional, Set, Tuple, Union
+
+    import numpy as np
+    import onnx
+    import onnxruntime  # type: ignore[import]
+    from onnxruntime.capi import _pybind_state as ORTC  # type: ignore[import]
+
+    # This is not use directly in DORT but needed by underlying exporter,
+    # so we still need to check if it exists.
+    importlib.import_module("onnxscript")
+
+    _NP_DTYPE = {
+        torch.float16: np.float16,
+        torch.float32: np.float32,
+        torch.float64: np.float64,
+        torch.uint8: np.uint8,
+        torch.int8: np.int8,
+        torch.int16: np.int16,
+        torch.int32: np.int32,
+        torch.int64: np.longlong,
+        torch.bool: np.bool_,
+    }
+    _ONNX_ELEMENT_TYPE_TO_TORCH_DTYPE = {
+        onnx.TensorProto.FLOAT: torch.float32,  # type: ignore[attr-defined]
+        onnx.TensorProto.FLOAT16: torch.float16,  # type: ignore[attr-defined]
+        onnx.TensorProto.DOUBLE: torch.float64,  # type: ignore[attr-defined]
+        onnx.TensorProto.BOOL: torch.bool,  # type: ignore[attr-defined]
+        onnx.TensorProto.UINT8: torch.uint8,  # type: ignore[attr-defined]
+        onnx.TensorProto.INT8: torch.int8,  # type: ignore[attr-defined]
+        onnx.TensorProto.INT16: torch.int16,  # type: ignore[attr-defined]
+        onnx.TensorProto.INT32: torch.int32,  # type: ignore[attr-defined]
+        onnx.TensorProto.INT64: torch.int64,  # type: ignore[attr-defined]
+    }
+    _TORCH_DTYPE_TO_ONNX_ELEMENT_TYPE = {
+        value: key for key, value in _ONNX_ELEMENT_TYPE_TO_TORCH_DTYPE.items()
+    }
```

Comment:
Can this be consolidated/reused at https://github.com/pytorch/pytorch/blob/main/torch/onnx/_internal/fx/type_utils.py?

Reply:
Yep, I will move them to `type_utils.py`.